### PR TITLE
Animations and UI modifications

### PR DESCRIPTION
**Pull request by pMalvasio**
* `TextFieldView` now change the layout based on states and not drawing extra views
* Animations for title and errors
* Some small UI changes for spacings

test

### CODE REVIEW

File: swift-ui-base/Common/View/TextField/Model/TextFieldData.swift [hunk 1]
```diff
@@ -11,16 +11,16 @@ import Foundation
...
   }
+  
   var validationType: ValidationType
   var errorMessage: String
   var title: String
   var isSecure = false
-  var hasTyped = false
-  var isValid = true
+
+  var isValid = true//@Published var isValid = true
```
Comment (glm4) [inline swift-ui-base/Common/View/TextField/Model/TextFieldData.swift:23]: remove comment if not used. Maybe open an issue to make these classes Combine friendly.

File: swift-ui-base/Common/View/TextField/Model/TextFieldData.swift [hunk 2]
```diff
@@ -11,16 +11,16 @@ import Foundation
 final class TextFieldData: ObservableObject {
   var value: String = "" {
     didSet {
-      hasTyped = hasTyped || !value.isEmpty
```
Comment (mato2593) [inline swift-ui-base/Common/View/TextField/Model/TextFieldData.swift:14]: Why did you remove this?

File: swift-ui-base/Common/View/TextField/Model/TextFieldData.swift (hunk 1 above)
Comment (mato2593) [inline swift-ui-base/Common/View/TextField/Model/TextFieldData.swift:23]: Do you need the commented var?

File: swift-ui-base/Common/View/TextField/View/TextFieldView.swift [hunk 3]
```diff
@@ -39,24 +33,23 @@ struct TextFieldView: View {
...
         .opacity(0.5)
       
-      if !fieldData.isValid && fieldData.hasTyped {
-        Text(fieldData.errorMessage)
-          .frame(maxWidth: .infinity, alignment: .leading)
-          .font(.footnote)
-          .foregroundColor(.errorRed)
-      }
+      Text(fieldData.errorMessage)
+        .frame(maxWidth: .infinity, alignment: .leading)
```
Comment (glm4) [inline swift-ui-base/Common/View/TextField/View/TextFieldView.swift:48]: wouldn't .infinity make the text to overflow if its long enough?

File: swift-ui-base/Common/View/TextField/Model/TextFieldData.swift (hunk 1 above)
Comment (pMalvasio) [reply swift-ui-base/Common/View/TextField/Model/TextFieldData.swift:23]: Need to remove that, was there because I was testing other stuff

Comment (pMalvasio) [reply swift-ui-base/Common/View/TextField/Model/TextFieldData.swift:23]: Agree, I only create this PR to change the way the textfield behaves based on states not to modify the model behind. That should be addressed in a different PR

Comment (germanStabile) [reply swift-ui-base/Common/View/TextField/Model/TextFieldData.swift:23]: how's it not combine friendly?

File: swift-ui-base/Common/View/TextField/View/TextFieldView.swift (hunk 3 above)
Comment (pMalvasio) [reply swift-ui-base/Common/View/TextField/View/TextFieldView.swift:48]: Yes, unles you have a padding removing space from the sides and that's the case of the wrapping stack

File: swift-ui-base/Common/View/TextField/View/TextFieldView.swift [hunk 4]
```diff
@@ -39,24 +33,23 @@ struct TextFieldView: View {
           TextField("", text: $fieldData.value)
             .foregroundColor(.darkGray)
             .autocapitalization(.none)
-            .opacity(0.8)
```
[the remark this answers is not in the document]
Comment (pMalvasio) [reply swift-ui-base/Common/View/TextField/View/TextFieldView.swift:42]: Actually, I don't remember deleting that line 🤔

[the remark this answers is not in the document]
Comment (germanStabile) [reply swift-ui-base/Common/View/TextField/View/TextFieldView.swift:42]: I blame your cat

File: swift-ui-base/Common/View/TextField/Model/TextFieldData.swift (hunk 2 above)
Comment (pMalvasio) [reply swift-ui-base/Common/View/TextField/Model/TextFieldData.swift:14]: Already discussed on slack 😉

File: swift-ui-base/Common/View/TextField/View/TextFieldView.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (pMalvasio) [reply swift-ui-base/Common/View/TextField/View/TextFieldView.swift:42]: Fair enough

File: swift-ui-base/Common/View/TextField/View/TextFieldView.swift (hunk 3 above)
Comment (glm4) [reply swift-ui-base/Common/View/TextField/View/TextFieldView.swift:48]: And how's the actual text behavior in that case, it would just clip it or we must add some modifier to the Text to show elipsis or similar?

Comment (pMalvasio) [reply swift-ui-base/Common/View/TextField/View/TextFieldView.swift:48]: Sincerely, I can even imagine your question on my mind right now. Let me try variations on code and I can answer you later

File: swift-ui-base/Common/View/TextField/View/TextFieldView.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (pMalvasio) [reply swift-ui-base/Common/View/TextField/View/TextFieldView.swift]: Added again 😉

File: swift-ui-base/Common/View/TextField/View/TextFieldView.swift (hunk 3 above)
Comment (pMalvasio) [reply swift-ui-base/Common/View/TextField/View/TextFieldView.swift:49]: @glm4, behavior for long string:

Screen Shot 2020-07-27 at 1 54 15 PM

Comment (glm4) [reply swift-ui-base/Common/View/TextField/View/TextFieldView.swift:49]: cool!
